samples/places-autocomplete-addressform/src/index.ts

```diff
@@ -56,44 +68,64 @@ function initAutocomplete() {
 function fillInAddress() {
   // Get the place details from the autocomplete object.
   const place = autocomplete.getPlace();
-
-  for (const component in componentForm) {
-    (document.getElementById(component) as HTMLInputElement).value = "";
-    (document.getElementById(component) as HTMLInputElement).disabled = false;
-  }
+  let address1 = "";
+  let postcode = "";
 
   // Get each component of the address from the place details,
   // and then fill-in the corresponding field on the form.
   for (const component of place.address_components as google.maps.GeocoderAddressComponent[]) {
     // @ts-ignore remove once typings fixed
     const addressType = component.types[0];
 
-    if (componentForm[addressType]) {
-      const val = component[componentForm[addressType]];
-      (document.getElementById(addressType) as HTMLInputElement).value = val;
+    switch (addressType) {
+      case "street_number": {
+        address1 = component[componentLength[addressType]] + " " + address1;
+        break;
+      }
+
+      case "route": {
+        address1 += component[componentLength[addressType]];
+        break;
+      }
+
+      case "postal_code": {
+        postcode = component[componentLength[addressType]] + postcode;
+        break;
+      }
+
+      case "postal_code_suffix": {
+        postcode += "-" + component[componentLength[addressType]];
+        break;
+      }
+
+      default: {
+        if (componentLength[addressType]) {
+          const val = component[componentLength[addressType]];
+          (document.getElementById(
+            addressType
+          ) as HTMLInputElement).value = val;
+        }
+        break;
+      }
     }
   }
-}
-// [END maps_places_autocomplete_addressform_fillform]
 
-// [START maps_places_autocomplete_addressform_geolocation]
-// Bias the autocomplete object to the user's geographical location,
-// as supplied by the browser's 'navigator.geolocation' object.
-function geolocate() {
-  if (navigator.geolocation) {
-    navigator.geolocation.getCurrentPosition((position) => {
-      const geolocation = {
-        lat: position.coords.latitude,
-        lng: position.coords.longitude,
-      };
-      const circle = new google.maps.Circle({
-        center: geolocation,
-        radius: position.coords.accuracy,
-      });
-      autocomplete.setBounds(circle.getBounds()!);
-    });
+  address1Field.value = address1;
+  postalField.value = postcode;
+
+  // Enable the rest of the address form fields
+  for (const component of componentFields) {
+    // @ts-ignore
+    (document.getElementById(component) as HTMLInputElement).disabled = false;
```

Comment:
I think these inputs should always be enabled. A best in class autocomplete able form will *also* play nice with browser based autocomplete (which, when available, has a higher likelihood of being accurate for the user - e.g. including apt #), and disabling the form inputs will block that.

additionally, using ids is generally not best practice, so this example isn't portable for the customer

Reply:
Ah, I remove the "disabled" attribute from the form fields after your feedback but forgot to remove this code which is now unecessary.